packages/wrangler/src/pages/build.tsx

```diff
@@ -172,15 +172,19 @@ export async function buildFunctions({
 
 	if (config.routes) {
 		if (!directory) {
-			logger.warn(
-				"Cannot write generated routes file _routes.generated.json. Output directory is undefined."
-			);
+			if (!isInPagesCI) {
+				logger.warn(
+					"Cannot write generated routes file _routes.generated.json. Output directory is undefined."
+				);
+			}
 		} else if (existsSync(join(directory, "_routes.json"))) {
 			logger.log(
 				"Found _routes.json file. Skipping automatic routes generation."
 			);
 		} else {
-			const routesJSON = convertRoutesToRoutesJSONSpec(config.routes);
+			// Reverse routes so that the least-specific are at the top. This will make
+			// route consolidation more effecient, as well as improve speed on the Cloudflare edge.
+			const routesJSON = convertRoutesToRoutesJSONSpec(config.routes.reverse());
```

Comment:
This reversal seems like it should be the job of `convertRoutesToRoutesJSONSpec`. Also, a reversal is super implicit. Why are the least-specific currently at the end of the list? Maybe we should sort by specificity inside of `convertRoutesToRoutesJSONSpec`?

Reply:
Maybe so - but I was thinking we could get the sorting for free since it's already done inside `generateConfigFromFileTree()` https://github.com/cloudflare/wrangler2/blob/jhands/functions-routing-consolidation/packages/wrangler/src/pages/functions/filepath-routing.ts#L112

To me it felt a little weird to do the reversal inside `convertRoutesToRoutesJSONSpec`. But maybe explicit ordering would be best

Reply:
> But maybe explicit ordering would be best

Personally, I think so. But it's totally not blocking so, approved :)

Reply:
Made ordering explicit :)